libs/datasets/dataset_orchestrator.py

```diff
@@ -0,0 +1,318 @@
+        region_overrides = json.load(open(REGION_OVERRIDES_JSON))
+        return MultiRegionOrchestrator(
+            regions=regions,
+            region_overrides=region_overrides,
+            cbsa_aggregator=cbsa_aggregator,
+            print_stats=print_stats,
+        )
+
+    def build_and_combine_regions(
+        self, aggregate_to_country: bool = True, generate_cbsas: bool = True
+    ):
+        processed_regions = parallel_map(self._build_region_timeseries, self.regions)
+        _log.info("Finished processing individual regions...")
+        return self._combine_regions(
+            list(processed_regions),
+            aggregate_to_country=aggregate_to_country,
+            generate_cbsas=generate_cbsas,
+        )
+
+    def _combine_regions(
+        self,
+        datasets: List[MultiRegionDataset],
+        aggregate_to_country: bool = True,
+        generate_cbsas: bool = True,
+        generate_hsas: bool = True,
+    ) -> MultiRegionDataset:
+        regions = [region for dataset in datasets for region in dataset.location_ids]
+        assert len(regions) == len(set(regions)), "Can't combine datasets with duplicate locations"
+
+        _log.info("Starting region combination...")
+        timeseries_df = (
+            pd.concat([dataset.timeseries_bucketed for dataset in datasets])
+            .sort_index()
+            .rename_axis(columns=PdFields.VARIABLE)
+        )
+        static_df = (
+            pd.concat(dataset.static for dataset in datasets)
+            .sort_index()
+            .rename_axis(columns=PdFields.VARIABLE)
+        )
+        tag = pd.concat([dataset.tag for dataset in datasets]).sort_index()
+        multiregion_dataset = MultiRegionDataset(
+            timeseries_bucketed=timeseries_df, static=static_df, tag=tag
+        )
+
+        if generate_hsas:
+            _log.info("Starting HSA aggregation...")
+            hsa_aggregator = statistical_areas.CountyToHSAAggregator.from_local_data()
+            multiregion_dataset = hsa_aggregator.aggregate(multiregion_dataset)
+
+        if generate_cbsas:
+            _log.info("Starting CBSA aggregation...")
+            cbsa_dataset = self.cbsa_aggregator.aggregate(
+                multiregion_dataset, reporting_ratio_required_to_aggregate=DEFAULT_REPORTING_RATIO
+            )
+            multiregion_dataset = multiregion_dataset.append_regions(cbsa_dataset)
+            multiregion_dataset.print_stats("CBSA dataset")
+
+        if aggregate_to_country:
+            _log.info("Starting country aggregation...")
+            multiregion_dataset = custom_aggregations.aggregate_to_country(
+                multiregion_dataset, reporting_ratio_required_to_aggregate=DEFAULT_REPORTING_RATIO
+            )
+            multiregion_dataset.print_stats("Aggregate to country")
+        return multiregion_dataset
+
+    def _build_region_timeseries(self, region_dataset: OneStateDataset):
```

Comment:
nit: Perhaps name this state_dataset if you rename the other variable to state_datasets.